File: iris.R
# Load the Iris dataset into a data frame
iris_df <- read.csv("/path/to/iris.csv", header = TRUE)

# Calculate the mean, median, and standard deviation of each variable for each species
species_means <- aggregate(iris_df[,1:4], by = list(species = iris_df$Species), FUN = mean)
species_medians <- aggregate(iris_df[,1:4], by = list(species = iris_df$Species), FUN = median)
species_stddevs <- aggregate(iris_df[,1:4], by = list(species = iris_df$Species), FUN = sd)

# Combine the mean, median, and standard deviation into a single data frame
species_stats <- merge(species_means, species_medians, by = "species")
species_stats <- merge(species_stats, species_stddevs, by = "species", suffixes = c("_mean", "_median", "_stddev"))

# Calculate the correlation matrix between each pair of variables
cor_matrix <- cor(iris_df[,1:4])

# Create a heatmap of the correlation matrix
library(ggplot2)
library(reshape2)
melted_cor_matrix <- melt(cor_matrix)
ggplot(melted_cor_matrix, aes(x = variable, y = variable2, fill = value)) + geom_tile() + scale_fill_gradient2(low = "red", mid = "white", high = "blue", midpoint = 0) + theme(axis.text.x = element_text(angle = 90, hjust = 1))

# Create a scatter plot matrix of the variables
pairs(iris_df[,1:4], main = "Scatter Plot Matrix of Iris Variables")

# Calculate the mean sepal length for each species and compare to the overall mean
mean_sepal_length_by_species <- aggregate(iris_df$Sepal.Length, by = list(species = iris_df$Species), FUN = mean)
overall_mean_sepal_length <- mean(iris_df$Sepal.Length)
mean_sepal_length_by_species$difference_from_overall_mean <- mean_sepal_length_by_species$x - overall_mean_sepal_length
mean_sepal_length_by_species

File: iris.py
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

# Load the Iris dataset into a pandas DataFrame
iris_df = pd.read_csv('/path/to/iris.csv')

# Calculate the mean, median, and standard deviation of each variable for each species
species_means = iris_df.groupby('Species').mean()
species_medians = iris_df.groupby('Species').median()
species_stddevs = iris_df.groupby('Species').std()

# Combine the mean, median, and standard deviation into a single DataFrame
species_stats = pd.concat([species_means, species_medians, species_stddevs], axis=1, keys=['Mean', 'Median', 'Std Dev'])

# Calculate the correlation matrix between each pair of variables
cor_matrix = iris_df.corr()

# Create a heatmap of the correlation matrix
sns.heatmap(cor_matrix, annot=True, cmap='coolwarm')

# Create a scatter plot matrix of the variables
sns.pairplot(iris_df, hue='Species')

# Calculate the mean sepal length for each species and compare to the overall mean
mean_sepal_length_by_species = iris_df.groupby('Species')['Sepal.Length'].mean()
overall_mean_sepal_length = iris_df['Sepal.Length'].mean()
mean_sepal_length_by_species_difference_from_overall_mean = mean_sepal_length_by_species - overall_mean_sepal_length
mean_sepal_length_by_species
